src/ts/components/core/Tree.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+
+interface Props extends BoxProps, StylesApiProps, DashBaseProps {
+    /** Determines whether tree nodes range can be selected with click when Shift key is pressed, `true` by default */
+    allowRangeSelection?: boolean;
+    /** Determines if checkboxes should be rendered, `false` by default */
+    checkboxes?: boolean;
+    /** Determines checked nodes as a list of values (note that only leaves can be checked), `[]` by default */
+    checked?: string[];
+    /** Determines whether tree node should be checked on space key press, `false` by default */
+    checkOnSpace?: boolean;
+    /** Determines whether selection should be cleared when user clicks outside of the tree, `false` by default */
+    clearSelectionOnOutsideClick?: boolean;
+    /** Data used to render nodes */
+    data: TreeNodeData[];
+    /** Determines expanded nodes as a list of values or `'*'` for all, `[]` by default */
+    expanded?: string[] | "*";
+    /** Determines whether tree node with children should be expanded on click, `true` by default */
+    expandOnClick?: boolean;
+    /** Determines whether tree node with children should be expanded on space key press, `true` by default */
+    expandOnSpace?: boolean;
+    /** Horizontal padding of each subtree level, key of `theme.spacing` or any valid CSS value, `'lg'` by default */
+    levelOffset?: string | number;
+    /** Determines selected nodes as a list of values, `[]` by default */
+    selected?: string[];
+    /** Determines whether node should be selected on click, `false` by default */
+    selectOnClick?: boolean;
+}
+
+const Leaf = ({
+    node,
+    expanded,
+    hasChildren,
+    elementProps,
+    tree,
+    checkboxes,
+    expandOnClick,
+}: RenderTreeNodePayload & { checkboxes: boolean; expandOnClick: boolean }) => {
+    const checked = tree.isNodeChecked(node.value);
+    const indeterminate = tree.isNodeIndeterminate(node.value);
+    return (
+        <Group gap="xs" {...elementProps}>
+            {checkboxes && (
+                <Checkbox.Indicator
+                    checked={checked}
+                    indeterminate={indeterminate}
+                    onClick={() =>
+                        !checked
+                            ? tree.checkNode(node.value)
+                            : tree.uncheckNode(node.value)
+                    }
+                />
+            )}
+            <Group
+                gap={5}
+                onClick={
+                    expandOnClick
+                        ? () => tree.toggleExpanded(node.value)
+                        : undefined
+                }
+            >
+                <span>{node.label}</span>
+                {hasChildren && (
+                    <AccordionChevron
+                        style={{
+                            transform: expanded
+                                ? "rotate(180deg)"
+                                : "rotate(0deg)",
```

Comment:
I know this is how the Mantine docs do it, but I find this very confusing. The standard behavior, that you can see even right here in GitHub, is that a collapsed node has the chevron pointing to the right, and an expanded node has it pointing down.
<img width="373" alt="Screenshot 2025-01-17 at 09 16 10" src="https://github.com/user-attachments/assets/86a0dc46-28fa-4461-a5fd-76fdd7008106" />
Which I think would be this:
```suggestion
                            transform: expanded
                                ? "rotate(0deg)"
                                : "rotate(270deg)",
```

Reply:
Nice catch - I found those chevrons confusing but couldn't say why. lol

Part of the bigger issue with wrapping this component for Dash is how to customize things like this chevron, and formatting the Tree to include other components like icons.  Even this [forum topic](https://community.plotly.com/t/file-explorer-tree-generator-for-local-files/68732/) had a debate on whether the chevron should be to the right or left of the text....  

If you were creating an app using React and Mantine, then it's easy to customize all of these things.  But how can we design an API for Dash to allow this flexibility?

Reply:
Right, that's always a challenge when going from a relatively low-level interface like Mantine to a higher-level interface like Dash components. Generally I'd say you _don't_ design for all the flexibility Mantine gives you; you make smart choices for the defaults and then add flexibility where a significant number of users will want it.

And "which side are the chevrons on" might be just such a case - personally I prefer them on the left, and given that that's where Mac, Windows, and GH all put them I think that would be a solid default, but since others like the right side we could certainly add a prop like `chevronSide = "left" | "right" | "none"`

One note though about putting them on the left: leaf nodes don't get a chevron, but we should probably add blank space as if they did, otherwise leaves won't look like they're really inside their parent folders.

Reply:
Good point about the icon not necessarily being a chevron - so `chevronSide` would be limiting (also maybe not the most well-known word!). But do you ever see users wanting changeable icons on both sides? That feels weird to me, seems like the icon that changes when the node is opened or closed will either be on the left or right, and if you want an _unchanging_ icon on the other side you can presumably put that in the `label`?

If we agree on that, then we can have maybe `expandIconSide = "left" | "right" | "none"`?

Then to the question of changing the icons: The advantage of a CSS rotation is you can animate it (which means we should use `-90deg` rather than `270deg` as I had suggested above). Would animation still work if you specified the same icon with different CSS? Something like:
```
collapsedIcon=DashIconify(icon="tabler:chevron-down", style={"transform": "rotate(-90deg)", "transition": "transform 0.5s"})
expandedIcon=DashIconify(icon="tabler:chevron-down", style={"transform": "rotate(0deg)", "transition": "transform 0.5s"})
```
If that preserves the animation, it seems like a good compromise, covering the vast majority of use cases without bloating the API too much.

Reply:
Changing the icons based on state, could be done like the checkbox with the checked or intermediate states:
https://www.dash-mantine-components.com/components/checkbox#change-icons

Reply:
> Good point about the icon not necessarily being a chevron - so `chevronSide` would be limiting (also maybe not the most well-known word!). But do you ever see users wanting changeable icons on both sides? That feels weird to me, seems like the icon that changes when the node is opened or closed will either be on the left or right, and if you want an _unchanging_ icon on the other side you can presumably put that in the `label`?
> 
> If we agree on that, then we can have maybe `expandIconSide = "left" | "right" | "none"`?
> 
> Then to the question of changing the icons: The advantage of a CSS rotation is you can animate it (which means we should use `-90deg` rather than `270deg` as I had suggested above). Would animation still work if you specified the same icon with different CSS? Something like:
> 
> ```
> collapsedIcon=DashIconify(icon="tabler:chevron-down", style={"transform": "rotate(-90deg)", "transition": "transform 0.5s"})
> expandedIcon=DashIconify(icon="tabler:chevron-down", style={"transform": "rotate(0deg)", "transition": "transform 0.5s"})
> ```
> 
> If that preserves the animation, it seems like a good compromise, covering the vast majority of use cases without bloating the API too much.

Animation does indeed still work when specifying the same icon. I have updated the PR to include this functionality.